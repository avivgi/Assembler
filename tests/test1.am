.define sz = 2
MAIN: mov r3, LIST[sz]
LOOP: jmp L1
prn #-5
mov STR[5], STR[2]
sub r1, r4
cmp r4, #sz
bne END
L1: bne LOOP
END: hlt
.define len = 4
STR: .string "abcdef"
LIST: .data 6, -9, len
cmp r6, #sz
cmp r7, #sz
cmp r6, #sz
cmp r7, #sz
K: .data 22
