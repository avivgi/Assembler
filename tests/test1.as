.define sz = 2
MAIN: mov r3, LIST[sz]
LOOP: jmp L1
mcr m_mcr1
cmp r4, #sz
bne END
endmcr
prn #-5
mov STR[5], STR[2]
sub r1, r4
m_mcr1
L1: bne LOOP
END: hlt
.define len = 4
mcr m_mcr2
cmp r6, #sz
cmp r7, #sz
endmcr
STR: .string "abcdef"
LIST: .data 6, -9, len
K: .data 22
m_mcr2
m_mcr2
K: .data 22
