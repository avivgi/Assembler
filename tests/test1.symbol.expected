Symbol name: sz        	Value: 2	type: 1: MDEFINE
Symbol name: MAIN      	Value: 100	type: 2: CODE
Symbol name: LOOP      	Value: 100	type: 2: CODE
Symbol name: L1        	Value: 100	type: 2: CODE
Symbol name: END       	Value: 100	type: 2: CODE
Symbol name: len       	Value: 4	type: 1: MDEFINE
Symbol name: STR       	Value: 100	type: 3: DATA
Symbol name: LIST      	Value: 107	type: 3: DATA
Symbol name: K         	Value: 110	type: 3: DATA
